test/Kestrel.Transport.FunctionalTests/ChromeTests.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#if NETCOREAPP2_2
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+using Microsoft.AspNetCore.Server.Kestrel.FunctionalTests;
+using Microsoft.AspNetCore.Testing;
+using Microsoft.AspNetCore.Testing.xunit;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Testing;
+using Xunit;
+
+namespace Interop.FunctionalTests
+{
+    [SkipIfChromeUnavailable]
+    public class ChromeTests : LoggedTest
+    {
+        private static readonly string _postHtml =
+@"<!DOCTYPE html>
+<html>
+ <head>
+   <script type=""text/javascript"">
+    function dosubmit() { document.forms[0].submit(); }
+   </script>
+ </head>
+ <body onload=""dosubmit();"">
+   <form action=""/"" method=""POST"" accept-charset=""utf-8"">
+   </form>
+ </body>
+</html>";
+
+        private string NetLogPath { get; set; }
+        private string StartupLogPath { get; set; }
+        private string ShutdownLogPath { get; set; }
+        private string ChromeArgs { get; set; }
+
+        private void InitializeArgs()
```

Comment:
This is a little silly.

Ideally, I would like to do this in an override of the Initialize(...) method of the base LoggedTest class, since Initialize must be called first to resolve the log output directory and the test method name for logging purposes. However, all overridden methods are public and xunit complains that all public methods must be marked as Theory. As far as I can tell, there's no way to turn this off but https://github.com/xunit/xunit/issues/1611 notes that in newer versions of the analyzer, you can create an attribute with a name that has the suffix `IgnoreXunitAnalyzersRule1013Attribute` to ignore this analyzer error. Alternatively, I can add a theory attribute with a skip string but that also looks silly.

Reply:
Alternatively you could put this into a ChromeTestBase class